tests/MongoDB.Driver.Core.Tests/Core/Clusters/DnsClientWrapperTests.cs

```diff
@@ -156,22 +147,34 @@ public void ResolveTxtRecords_should_throw_when_domainName_is_null(bool async)
         [InlineData(true)]
         public void ResolveTxtRecords_should_throw_when_cancellation_is_already_requested(bool async)
         {
-            var subject = new DnsClientWrapper();
+            var subject = CreateSubject();
             var domainName = "test5.test.build.10gen.cc";
             var cts = new CancellationTokenSource();
             cts.Cancel();
 
             Exception exception;
             if (async)
             {
-                exception = Record.Exception(() => subject.ResolveTxtRecords(domainName, cts.Token));
+                exception = Record.Exception(() => subject.ResolveTxtRecordsAsync(domainName, cts.Token).GetAwaiter().GetResult());
             }
             else
             {
-                exception = Record.Exception(() => subject.ResolveTxtRecordsAsync(domainName, cts.Token).GetAwaiter().GetResult());
+                exception = Record.Exception(() => subject.ResolveTxtRecords(domainName, cts.Token));
             }
 
             exception.Should().Match<Exception>(e => e is OperationCanceledException || e.InnerException is OperationCanceledException);
         }
+
+        // private methods
+        private IDnsResolver CreateSubject()
+        {
+            DnsClientWrapperReflector.__instance(null); // reset cache
+            return DnsClientWrapper.Instance;
+        }
```

Comment:
Every test should start with a fresh instance of `DnsClientWrapper`. Since it is now a singleton there is no public API to create new instances, but we can do it using reflection:

    private DnsClientWrapper CreateSubject()
    {
        var constructorInfo = typeof(DnsClientWrapper).GetConstructors(BindingFlags.NonPublic | BindingFlags.Instance).Single();
        var subject = (DnsClientWrapper)constructorInfo.Invoke(new object[0]);
        return subject;
    }

Reply:
done